common/lib/xmodule/xmodule/course_module.py

```diff
@@ -178,6 +177,141 @@ def to_json(self, values):
                 continue
         return json_data
 
+class ProctoringConfiguration(Dict):
+    def from_json(self, value):
+        """
+        Return ProctoringConfiguration as full featured Python type. Perform validation on the backend
+        and the rules and include any inherited values from the platform default.
+        """
+        errors = []
+        value = super(ProctoringConfiguration, self).from_json(value)
+        proctoring_backend_settings = getattr(
+            settings,
+            'PROCTORING_BACKENDS',
+            None
+        )
+
+        backend_errors = self._validate_proctoring_backend(value, proctoring_backend_settings)
+        rules_errors = self._validate_proctoring_rules(value, proctoring_backend_settings)
+
+        errors.extend(backend_errors)
+        errors.extend(rules_errors)
+
+        if errors:
+            raise ValueError(errors)
+
+        value = self._get_proctoring_value(value, proctoring_backend_settings)
+
+        return value
+
+    def _get_proctoring_value(self, value, proctoring_backend_settings):
+        """
+        Return a proctoring value that includes any inherited attributes from the platform defaults
+        for the backend or rules.
+        """
+        proctoring_provider = value.get('backend', None)
+        proctoring_provider_rules = value.get('rules', None)
+
+        # if both are missing from the value, return the default
+        if proctoring_provider is None and proctoring_provider_rules is None:
+            return self.default
+
+        # if provider is missing, but rules are not, use the default provider
+        # TODO: this would cause a mismatch between the provider and the rules...
```

Comment:
TODO: We need to decide on what to do in this case. If you have a provider set with rules associated with it, and try to switch to a provider that has no rules, you will get a validation error. Is this acceptable? Or do we want to blow away any rules automatically in favor of the platform defaults?